### ISSUE TITLE
Add req-based ship spawning

### ISSUE THREAD

**Issue by Choosechee** [FTL-Hyperspace/FTL-Hyperspace#129]
Localresources was trying to make more bosses be affected by challenge levels, but the only way to do that is by having four event copies for each boss, one for each level. This bloats the XML, so Jersee postponed the idea. I want more bosses to be affected by challenge levels, so I propose being able to load a ship based on a requirement. This would make it much simpler and lighter weight to implement the idea.

**Comment by Nasa62** [comment]
At some point we intend to allow loading a blueprint from Lua, this depends on #101 first and then we can start exploring that option.
